database/import.go

```diff
@@ -102,44 +104,79 @@ func dropHeadersFile(tmpHeadersFile *os.File, tmpHeadersFilePath string, log *ze
 	}
 }
 
-func parseRecord(record []string, rowIndex int32, previousBlockHash string) dto.DbBlockHeader {
-	hash := parseChainHash(record[0])
-	version := parseInt(record[1])
-	merkleroot := parseChainHash(record[2])
-	nonce := parseInt(record[3])
-	bits := parseInt(record[4])
-	chainWork := parseBigInt(record[5])
-	timestamp := parseInt64(record[6])
-	cumulatedWork := parseBigInt(record[7])
+func PrepareRecord(record []string, previousBlockHash string, bh service.BlockHasher, cumulatedChainWork string, rowIndex int) (dto.DbBlockHeader, error) {
+	parsedRow, err := parseRecordToBlockHeadersSource(record, previousBlockHash)
+	if err != nil {
+		return dto.DbBlockHeader{}, err
+	}
+	cumulatedChainWorkBigInt := parseBigInt(cumulatedChainWork)
+	rowIndexInt32 := int32(rowIndex)
+	preparedRecord := calculateFields(bh, parsedRow, cumulatedChainWorkBigInt, rowIndexInt32)
+	return preparedRecord, nil
+}
+
+func parseRecordToBlockHeadersSource(record []string, previousBlockHash string) (domains.BlockHeaderSource, error) {
+	if len(record) != 5 {
```

Comment:
This '5' should be calculated from the number of columns in the db, or at least a const somewhere on top of the file, to avoid magic numbers

Reply:
added const to store number of columns in csv file